skimage/morphology/rolling_ball.py

```diff
@@ -0,0 +1,178 @@
+
+    Returns
+    -------
+    filtered_image : ndarray
+        The image with background removed.
+    background : ndarray
+        The background that was removed.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from skimage import data
+    >>> from skimage.morphology import rolling_ball
+
+    Subtract a black background from an example image
+
+    >>> image = data.coins()
+    >>> result, bg = rolling_ball(image, radius=100)
+
+    Subtract a white background from an example image
+
+    >>> image = data.page()
+    >>> result, bg = rolling_ball(image, radius=100, white_background=True)
+
+    References
+    ----------
+    .. [1] Sternberg, Stanley R. "Biomedical image processing." Computer 1
+           (1983): 22-34. :DOI:`10.1109/MC.1983.1654163`
+    """
+
+    # TODO: list and other iterables
+    if isinstance(radius, tuple):
+        if not len(radius) == 2:
+            raise ValueError(f"Radius as typle needs to have length 2.")
+        space_vertex = radius[0]
+        intensity_vertex = radius[1]
+    else:
+        space_vertex = radius
+        intensity_vertex = radius
+
+    try:
+        space_vertex = float(space_vertex)
+    except ValueError:
+        raise ValueError(
+            f"Radius should be float or float tuple, was {type(space_vertex)}")
+
+    try:
+        intensity_vertex = float(intensity_vertex)
+    except ValueError:
+        raise ValueError(
+            f"Radius should be float or float tuple, "
+            f"was {type(intensity_vertex)}")
+
+    if space_vertex <= 0:
+        raise ValueError(
+            f"Spacial radius must be greater zeros, was {space_vertex}")
+
+    if intensity_vertex <= 0:
+        raise ValueError(
+            f"Intensity radius must be greater zeros, was {intensity_vertex}")
+
+    try:
+        white_background = bool(white_background)
+    except ValueError:
+        raise ValueError(
+            f"white_background must be convertible to a boolean, "
+            f"was {type(white_background)}"
+        )
+
+    if not isinstance(image, np.ndarray):
+        raise ValueError(
+            f"input_img must be a np.ndarray, "
+            f"was {type(image)}"
+        )
+
+    if not issubclass(image.dtype.type, np.integer):
+        raise ValueError("Currently only integer images are supported.")
+
+    img = image.copy()
+    if white_background:
+        img = invert(img)
+
+    # assuming selem touches the umbra at (x,y) pre-compute the
+    # (relative) height of selem at the center
+    # (selem is a ball/spheroid)
+    L = np.arange(-space_vertex, space_vertex + 1)
+    X, Y = np.meshgrid(L, L)
+    distance = np.sqrt(X ** 2 + Y ** 2)
+    sagitta = space_vertex - space_vertex * np.sqrt(
+        np.clip(1 - (distance ** 2 / intensity_vertex ** 2), 0, None)
+    )
+
+    kernel = np.array(distance <= space_vertex, dtype=float)
+    kernel[kernel == 0] = np.Inf
+
+    kernel_size = np.array(kernel.shape)
+    img_original = img.copy()
```

Comment:
Why this copy again? Can't we use `image` directly here?

Reply:
This has been refactored in firefoxmetzger/scikit-image@ee27cf6c467e1a03d4d4add2cccc5c2f47a1947b